tpm cmds: adding check for emtpy buffer

dpsmith opened the pull request:
The commands tpm2_pcr_extend and tpm1_pcr_extend expects to be passed a tpm instance that has an allocated buffer. As such this adds a check to ensure this expectation is met.

Signed-off-by: Daniel P. Smith <dpsmith@apertussolutions.com>

dpsmith committed:
tpm2_cmds: adding check for emtpy buffer

The command tpm2_extend expects to be passed a tpm instance that has an
allocated buffer. As such this adds a check to ensure this expectation
is met.

Signed-off-by: Daniel P. Smith <dpsmith@apertussolutions.com>

rossphilipson commented on tpm2_cmds.c:
```diff
@@ -97,6 +97,11 @@ int tpm2_extend_pcr(struct tpm *t, u32 pcr,
 	u16 size;
 	int ret = 0;
 
+	if (b == NULL) {
```
You should add this to tpm1_cmds.c too

dpsmith replied:
Will do! Will make it a separate commit.

dpsmith committed:
tpm1_cmds: adding check for emtpy buffer

The command tpm1_extend expects to be passed a tpm instance that has an
allocated buffer. As such this adds a check to ensure this expectation
is met.

Signed-off-by: Daniel P. Smith <dpsmith@apertussolutions.com>

dpsmith committed:
tpm_buff: add missing include for linux

The include <string.h> is missing from tpm_buff that results in a
warning during compilation.

Signed-off-by: Daniel P. Smith <dpsmith@apertussolutions.com>

rossphilipson reviewed (approved):
Reviewed-by: Ross Philipson <ross.philipson@oracle.com>
